url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/com/jgraph/layout/tree/OrganizationalChart.java

```diff
@@ -0,0 +1,341 @@
+					widestAmount = childWidth;
+					widestIndex = index;
+				}
+			}
+			if (widestIndex >= 0) {
+				StandardTreeNode sub = (StandardTreeNode) node.children.get(widestIndex);
+				sub.leftContour.dx = 0;
+				sub.rightContour.dx = (sub.width - node.width) / 2 + verticalEdgeLeftInset + verticalEdgeRightInset;
+
+				node.leftContour.next = sub.leftContour;
+				node.rightContour.next = sub.rightContour;
+			} else {
+				StandardTreeNode sub = (StandardTreeNode) node.children.get(0);
+
+				sub.leftContour.dx = (sub.width - node.width) / 2;
+				sub.rightContour.dx = (sub.width - node.width) / 2;
+
+				node.leftContour.next = sub.leftContour;
+				node.rightContour.next = sub.rightContour;
+			}
+			
+		} else if (node.children.size() == 1) {
+			StandardTreeNode sub = (StandardTreeNode) node.children.get(0);
+			sub.setDepth(node.getDepth() + 1);
+			layout(sub);
+
+			sub.leftContour.dx = (sub.width - node.width) / 2;
+			sub.rightContour.dx = (sub.width - node.width) / 2;
+
+			node.leftContour.next = sub.leftContour;
+			node.rightContour.next = sub.rightContour;
+		} else {
+			for (Iterator it = node.children.iterator(); it.hasNext();) {
+				StandardTreeNode n = (StandardTreeNode) it.next();
+				n.setDepth(node.getDepth() + 1);
+				layout(n);
+			}
+
+			join(node);
+		}
+	}
+	
+	/**
+	 * An ADT representing a node in an organisational chart structure.
+	 */
+	protected class OrganizationalTreeNode extends StandardTreeNode {
+		
+		/** Whether or not children of this node this have vertical children */
+		protected boolean verticalStyleChildren = false;
+		
+		
+		public OrganizationalTreeNode(Object cell) {
+			super(cell);
+			// Check if this cell is in the explicit set of vertical vertices
+			if (horizontalParentsSet != null && horizontalParentsSet.contains(cell)) {
+				verticalStyleChildren = true;
+			}
+		}
+
+		/**
+		 * Adds a new child to this parent node
+		 * @param newChild the child to be added
+		 */
+		public void addChild(StandardTreeNode newChild) {
+			children.add(newChild);
+			if (children.size() >= childrenLimitOrientationSwitch) {
+				verticalStyleChildren = true;
+			}
+		}
+		
+		/**
+		 * @return the verticalStyleChildren
+		 */
+		public boolean isVerticalStyleChildren() {
+			return verticalStyleChildren;
+		}
+
+		/**
+		 * @param verticalStyleChildren the verticalStyleChildren to set
+		 */
+		public void setVerticalStyleChildren(boolean verticalStyleChildren) {
+			this.verticalStyleChildren = verticalStyleChildren;
+		}
+		
+		/**
+		 * @param depth the depth to set
+		 */
+		public void setDepth(int depth) {
+			super.setDepth(depth);
+			if (depth >= vertexDepthOrientationSwitch) {
+				verticalStyleChildren = true;
+			}
+		}
+		
+		/**
+		 * Sets the position of this node
+		 * 
+		 * @param parent
+		 *            the parent of this node
+		 * @param levelHeight
+		 *            the height of nodes on the same level
+		 */
+		public void setPosition(Point2D parent, double levelHeight) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 252-308 into a separate method.